src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.hpp

```
@@ -195,5 +197,89 @@ void damped_harmonic(
+                 const std::array<int, 3>& exps);
+
+  DampedHarmonic() = default;
+  DampedHarmonic(const DampedHarmonic&) = default;
+  DampedHarmonic& operator=(const DampedHarmonic&) = default;
+  DampedHarmonic(DampedHarmonic&&) = default;
+  DampedHarmonic& operator=(DampedHarmonic&&) = default;
+  ~DampedHarmonic() override = default;
+
+  /// \cond
+  explicit DampedHarmonic(CkMigrateMessage* msg);
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(DampedHarmonic);  // NOLINT
+  /// \endcond
+
+  std::unique_ptr<GaugeCondition> get_clone() const override;
+
+  template <size_t SpatialDim>
+  void gauge_and_spacetime_derivative(
+      gsl::not_null<tnsr::a<DataVector, SpatialDim, Frame::Inertial>*> gauge_h,
+      gsl::not_null<tnsr::ab<DataVector, SpatialDim, Frame::Inertial>*>
+          d4_gauge_h,
+      const Scalar<DataVector>& lapse,
+      const tnsr::I<DataVector, SpatialDim, Frame::Inertial>& shift,
+      const tnsr::a<DataVector, SpatialDim, Frame::Inertial>&
+          spacetime_unit_normal_one_form,
+      const Scalar<DataVector>& sqrt_det_spatial_metric,
+      const tnsr::II<DataVector, SpatialDim, Frame::Inertial>&
+          inverse_spatial_metric,
+      const tnsr::aa<DataVector, SpatialDim, Frame::Inertial>& spacetime_metric,
+      const tnsr::aa<DataVector, SpatialDim, Frame::Inertial>& pi,
+      const tnsr::iaa<DataVector, SpatialDim, Frame::Inertial>& phi,
+      double time,
+      const tnsr::I<DataVector, SpatialDim, Frame::Inertial>& inertial_coords)
+      const;
+
+  // NOLINTNEXTLINE(google-runtime-references)
+  void pup(PUP::er& p) override;
+
+ private:
+  double spatial_decay_width_{std::numeric_limits<double>::signaling_NaN()};
+  std::array<double, 3> amplitudes_{
+      {std::numeric_limits<double>::signaling_NaN(),
+       std::numeric_limits<double>::signaling_NaN(),
+       std::numeric_limits<double>::signaling_NaN()}};
+  std::array<int, 3> exponents_{{2, 2, 2}};
```

Since `int` doesn't have a `signaling_NaN`, is there a way to ensure that components have been set properly? Not default initializing `exponents_`? Just want to protect against a silent failure where the exponents somehow don't get initialized properly to their input values and are just 2.

---

Not that I trivially know of. I could set it to the maximum value. We do that for `size_t` and most things will break horribly in that case. Would that be work for you?

---

Yeah that sounds good!